sktime/classification/interval_based/_tsf.py

```diff
@@ -97,6 +109,66 @@ def __init__(self,
         # We need to add is-fitted state when inheriting from scikit-learn
         self._is_fitted = False
 
+    @classmethod
+    def _get_intervals(
+        cls, n_intervals, min_interval, series_length, random_state=None
+    ):
+        """
+        Generate random intervals for given parameters.
+        """
+        rng = check_random_state(random_state)
+        intervals = np.zeros((n_intervals, 2), dtype=int)
+        for j in range(n_intervals):
+            intervals[j][0] = rng.randint(series_length - min_interval)
+            length = rng.randint(series_length - intervals[j][0] - 1)
+            if length < min_interval:
+                length = min_interval
+            intervals[j][1] = intervals[j][0] + length
+        return intervals
+
+    @classmethod
+    def _get_transformed_data(cls, X, intervals):
+        """
+        Compute the mean, standard deviation and slope for given intervals
+        of input data X.
+        """
+        n_instances, _ = X.shape
+        n_intervals, _ = intervals.shape
+        transformed_x = np.empty(shape=(3 * n_intervals, n_instances), dtype=np.float32)
+        for j in range(n_intervals):
+            X_slice = X[:, intervals[j][0]: intervals[j][1]]
+            means = np.mean(X_slice, axis=1)
+            std_dev = np.std(X_slice, axis=1)
+            slope = cls._lsq_fit(X_slice)
+            transformed_x[3 * j] = means
+            transformed_x[3 * j + 1] = std_dev
+            transformed_x[3 * j + 2] = slope
+
+        return transformed_x.T
+
+    @classmethod
```

Comment:
See above, once we detach the class methods we can make this to a stand-alone function too.